### PR TITLE
Make node type for `waypoint` and `ztunnel` instead of using `sidecar`

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/42313

Ztunnel update is in https://github.com/istio/ztunnel/pull/314 (but not required)

**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 1]
```diff
@@ -1286,6 +1286,7 @@ func (cb *ClusterBuilder) getAllCachedSubsetClusters(clusterKey clusterCache) ([
 // build does any final build operations needed, like marshaling etc.
 func (mc *MutableCluster) build() *cluster.Cluster {
 	if mc == nil {
+		log.Errorf("howardjohn: nil MC")
```
Comment (dhawton) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:1289]: Is this needed?

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 2]
```diff
@@ -1299,9 +1300,12 @@ func (mc *MutableCluster) build() *cluster.Cluster {
 				},
 			}
 		}
+		log.Errorf("howardjohn: %v late set protocol", mc.cluster.Name)
```
Comment (dhawton) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:1303]: This, too

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 3]
```diff
@@ -1299,9 +1300,12 @@ func (mc *MutableCluster) build() *cluster.Cluster {
...
 		mc.cluster.TypedExtensionProtocolOptions = map[string]*anypb.Any{
 			v3.HttpProtocolOptionsType: protoconv.MessageToAny(mc.httpProtocolOptions),
 		}
+	} else {
+		log.Errorf("howardjohn: %v already set protocol: %+v", mc.cluster.Name)
```
Comment (dhawton) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:1308]: And this one

File: pilot/pkg/networking/core/v1alpha3/cluster_waypoint.go [hunk 4]
```diff
@@ -77,6 +77,7 @@ func (configgen *ConfigGeneratorImpl) buildWaypointInboundClusters(cb *ClusterBu
 
 func (cb *ClusterBuilder) buildWaypointInboundPodCluster(wl ambient.Workload, port model.Port) *MutableCluster {
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundPod, "", host.Name(wl.PodIP), port.Port)
+	log.Errorf("howardjohn: for %v build %v", cb.proxy.ID, clusterName)
```
Comment (dhawton) [inline pilot/pkg/networking/core/v1alpha3/cluster_waypoint.go:80]: Another here

File: pilot/pkg/xds/ads.go [hunk 5]
```diff
@@ -695,6 +695,7 @@ func (s *DiscoveryServer) computeProxyState(proxy *model.Proxy, request *model.P
 			}
 		}
 	}
+	log.Errorf("howardjohn: compute %v %v %v", sidecar, gateway, proxy.ID)
```
Comment (dhawton) [inline pilot/pkg/xds/ads.go:698]: One more

**Review by dhawton — COMMENTED** [review]
Not sure if you meant to leave these `howardjohn` log entries in

**Comment by howardjohn** [comment]
Fixed those, thanks.. I really need a git hook to stop that

**Review by dhawton — APPROVED** [review]
LGTM

**Pull request merged**